KarhooUISDK/Routing/MainFlowCoordinator.swift

```diff
@@ -0,0 +1,252 @@
+//
+//  BookingFlowCoordinator.swift
+//  KarhooUISDK
+//
+//  Created by Diana Petrea on 04.04.2023.
+//  Copyright © 2023 Flit Technologies Ltd. All rights reserved.
+//
+
+import Foundation
+import KarhooSDK
+
+protocol MainFlowRouter {
+    func startBooking(
+        with journeyInfo: JourneyInfo?,
+        passengerDetails: PassengerDetails?,
+        filters: [QuoteListFilter]?,
+        bookingMetadata: [String: Any]?,
+        callback: ((BookingScreenResult) -> Void)?
+    )
+    
+    // TODO: Add any parameters necessary to open these screens
+    func routeToAllocationScreen()
+    func routeToRidesList()
+    func routeToTrackDriver()
+}
+
+protocol MainFlowBookingRouter {
```

Comment:
What is the difference between `RidePlanningRouter` and `MainFlowBookingRouter`? The naming is a bit unclear, we might need to change it somehow.

Reply:
Ok, I see `MainFlowBookingRouter` is sort of specialized protocol to deal with Booking-process navigation only. But do we need to separate it from the main routing interface? If I see correctly, this `MainFlowBookingRouter` is not used anyhow right now.

Reply:
Ok, let's go with your approach. We shall see how it's doing in practice, and if needed, we can always simplify it and merge two router protocols into one. 
Thnaks for the clarification :+1: